app/router.js

```diff
@@ -6,7 +6,10 @@ var Router = Ember.Router.extend({
   location: config.locationType,
   onTransition: function() {
     this.get('analytics').page();
-    this.get('flashMessages').clearMessages();
+
+    if(config.flashMessageDefaults.sticky) {
```

Comment:
Should this be the inverse? I'm not sure what `sticky` means, but I'd expect if flash messages are sticky they would not be cleared by route transitions.